src/app/api/utils.ts

```diff
@@ -44,4 +58,30 @@ export function getWorkSectionColour(workSectionId?: WorkSectionId): string {
             break;
     }
     return colour;
+}
+
+export function createShiftsFromShiftCreatorPayload (shiftInfo: ShiftCreationPayload): Partial<Shift>[] {
```

Comment:
I wonder if we could name this a little differently.  Right now, the implementation details are within the name of the method (i.e. ShiftCreatorPayload).  However, if we named it just `createShifts` it would become ambiguous with the API method.  Therefore, we want to make the distinction between using the API and just generating objects.  For this purpose, I would be inclined to putting this into a static class that helps delineate between the API / factory. For example:

```ts
export class ShiftFactory{
   static createShifts(shiftsConfig:ShiftCreationPayload) : Partial<Shift>[]{
   }
}
```
Then when its being used...

```ts
import {ShiftFactory} from './api/utils'

let newShifts = ShiftFactory.createShifts(payload);

```